src/lib.rs

```diff
@@ -57,22 +63,40 @@ type Result<T> = std::result::Result<T, Error>;
 
 pub trait Key<'a>: Into<ExplicitKey<'a>> + Send + Sync {}
 
-struct UnredactedKey<'a>(Arg<'a>);
+impl<'a> Key<'a> for ExplicitKey<'a> {}
 
-impl<'a> Key<'a> for UnredactedKey<'a> {}
+impl<'a> Key<'a> for &'a ExplicitKey<'a> {}
 
-impl<'a> From<UnredactedKey<'a>> for ExplicitKey<'a> {
-    fn from(val: UnredactedKey<'a>) -> Self {
-        Self {
-            redacted: val.0.clone(),
-            unredacted: val.0,
+impl<'a> From<&'a ExplicitKey<'a>> for ExplicitKey<'a> {
+    fn from(val: &'a ExplicitKey<'a>) -> Self {
+        ExplicitKey {
+            unredacted: Arg::Borrowed(val.unredacted.as_bytes()),
+            redacted: val.redacted.as_ref().map(|a| Arg::Borrowed(a.as_bytes())),
         }
     }
 }
 
+#[derive(Clone)]
+pub struct ExplicitKey<'a> {
+    unredacted: Arg<'a>,
+
+    /// Present iff different.
```

Comment:
nit: typo

Reply:
No this is just a common shorthand for "if and only if"

Reply:
Interesting thing to learn today! I would probably consider rewording it anyway.

Reply:
I think I'd like to keep it. This is a common phrasing in math proofs, which is the mindset I like to adopt in setting invariants. It comes up several times in the rust repo for example (although admittedly I don't see any examples in public docstrings). Stuff like this:

```
[slamb@slamb-workstation ~/git/rust-lang/rust]$ rg -w iff
src/tools/miri/src/sync.rs
239:                "invariant violation: lock_count == 0 iff the thread is unlocked"
266:                .expect("invariant violation: lock_count == 0 iff the thread is unlocked");

compiler/rustc_borrowck/src/type_check/input_output.rs
133:                    "Expected body to have yield_ty ({:?}) iff we have a UR yield_ty ({:?})",

compiler/rustc_borrowck/src/diagnostics/outlives_suggestion.rs
42:    /// Returns `true` iff the `RegionNameSource` is a valid source for an outlives

compiler/rustc_hir/src/hir.rs
1928:    // This is `Some(_)` iff there is an explicit user-specified `label

compiler/rustc_type_ir/src/lib.rs
447:    Covariant,     // T<A> <: T<B> iff A <: B -- e.g., function return type
448:    Invariant,     // T<A> <: T<B> iff B == A -- e.g., type of mutable cell
449:    Contravariant, // T<A> <: T<B> iff B <: A -- e.g., function param type

...
```